### PR TITLE
Test max accuracy instead of final accuracy

### PR DESCRIPTION
**Pull request by zcain117**
Checked that all 5 tests work as expected

### CODE REVIEW

**Comment by taylanbil** [comment]
any reason to keep all epoch accuracies around? why not just keep latest and best so far? and maybe print something @ 

```
    msg = "Epoch: {}, Mean Accuracy: {:.2f}%".format(epoch, accuracy)
    if best_so_far:
      msg += ' - best so far'
    print(msg)
```

**Comment by zcain117** [comment]
> any reason to keep all epoch accuracies around? why not just keep latest and best so far? and maybe print something @
> 
> ```
>     msg = "Epoch: {}, Mean Accuracy: {:.2f}%".format(epoch, accuracy)
>     if best_so_far:
>       msg += ' - best so far'
>     print(msg)
> ```

Good idea - switched to do this

File: test/test_train_cifar.py [hunk 1]
```diff
@@ -221,19 +221,24 @@ def test_loop_fn(model, loader, device, context):
...
   num_training_steps_per_epoch = train_dataset_len // (
       FLAGS.batch_size * num_devices)
+  max_accuracy = 0.0
   for epoch in range(1, FLAGS.num_epochs + 1):
     model_parallel(train_loop_fn, train_loader)
     accuracies = model_parallel(test_loop_fn, test_loader)
     accuracy = mean(accuracies)
-    print("Epoch: {}, Mean Accuracy: {:.2f}%".format(epoch, accuracy))
+    if accuracy > max_accuracy:
+      print('New best accuracy:')
```
Comment (dlibenzi) [inline test/test_train_cifar.py]: I'm not sure that print belongs there.

Comment (zcain117) [reply test/test_train_cifar.py]: Removed the 'new best accuracy' print statements from the 5 tests and switched to more succinct way of updating max_accuracy
